src/lib/registration/ui/abort_confirmation.rb

```diff
@@ -0,0 +1,42 @@
+require "yast"
+
+module Registration
+  module UI
+    # this class displays a popup for the user to confirm the abort
+    class AbortConfirmation
+      Yast.import "Popup"
+      Yast.import "Mode"
+
+      # displays a new popup
+      # @return [Boolean] true whether the user confirmed he/she wants to abort
+      def self.run
+        dialog = new
+        dialog.run
+      end
+
+      # the constructor
+      def initialize
+        textdomain "registration"
+      end
+
+      # displays the popup
+      # @return [Boolean] true whether the user confirmed he/she wants to abort
+      def run
+        if Yast::Mode.installation || Yast::Mode.repair
```

Comment:
The `repair` mode is obsolete, it's not used anymore (the repair module has been dropped).